### PR TITLE
dsg-2088/reset-page-on-search

### PR DESCRIPTION
**Pull request by SouradeepSaha**
# Basics

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [x] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?

Fix bug by resetting page number to 0 when search is clicked.

## Screenshot of the affected change

Before:

https://github.com/Nuvalence/dsgov-web/assets/46139736/f4420fed-fb16-4ea2-9a83-469d228c73b1

After:

https://github.com/Nuvalence/dsgov-web/assets/46139736/2d670492-7496-4fd4-8e14-bcbd93441550

## Checklist

- [x] I have updated the documentation accordingly
- [ ] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

**Review by odeljanin — COMMENTED** [review]
Looks good, we could also add unit tests to increase coverage

File: libs/agency/feature/admin/src/lib/components/schemas-list/schemas-list.component.html [hunk 1]
```diff
@@ -8,7 +8,7 @@ <h1 class="title">Schemas</h1>
       [suffixIcon]="searchBoxIcon"
       [type]="'text'"
       (clickIcon)="clearSearch()"
-      (keyup.enter)="getSchemasList()"
+      (keyup.enter)="handleSearch()"
```
Comment (davidyoung77) [inline libs/agency/feature/admin/src/lib/components/schemas-list/schemas-list.component.html:11]: this is good, we should add tests to confirm this functionality is working

**Review by davidyoung77 — COMMENTED** [review]
Implementation looks good, please add tests

File: libs/agency/feature/admin/src/lib/components/schemas-list/schemas-list.component.spec.ts [hunk 2]
```diff
@@ -241,6 +241,18 @@ describe('SchemasListComponent', () => {
...
+    it('should set pageNumber to 0 on search', () => {
+      const containerDebugElement = fixture.debugElement.query(By.css('nuverial-text-input'));
+      expect(containerDebugElement).toBeTruthy();
+
+      const searchText = 'some search text';
+      component.searchInput.setValue(searchText);
+      containerDebugElement.triggerEventHandler('keyup.enter', {});
+      fixture.detectChanges();
+
+      expect(fixture.componentInstance.pagingRequestModel.pageNumber).toEqual(0);
```
Comment (davidyoung77) [inline libs/agency/feature/admin/src/lib/components/schemas-list/schemas-list.component.spec.ts:253]: I can't tell that this test is doing what you think it is. whet is the pagenumber before you set the searchInput?

**Review by davidyoung77 — APPROVED** [review]
Nice work

**Pull request merged**